fl4health/datasets/skin_cancer/preprocess_skin.py

```diff
@@ -0,0 +1,281 @@
+# The following code is adapted from the preprocess_skin.py script
+# from the medical_federated GitHub repository by Seongjun Yang et al.
+# Paper: https://arxiv.org/abs/2207.03075
+# Code: https://github.com/wns823/medical_federated.git
+# - medical_federated/skin_cancer_federated/preprocess_skin.py
+
+import json
+import os
+from typing import Any, Dict
+
+import pandas as pd
+
+
+def save_to_json(data: Dict[str, Any], path: str) -> None:
+    with open(path, "w", encoding="utf-8") as file:
+        json.dump(data, file, indent="\t")
+
+
+if __name__ == "__main__":
+
+    data_path = "fl4health/datasets/skin_cancer"
+
+    # Load GT file of ISIC-19
+    ISIC_2019_path = f"{data_path}/ISIC_2019"
+    ISIC_csv_path = os.path.join(ISIC_2019_path, "ISIC_2019_Training_GroundTruth.csv")
+    ISIC_df = pd.read_csv(ISIC_csv_path)
+
+    # Load metadata file of HAM10000
+    HAM_10000_path = f"{data_path}/HAM10000"
+    HAM_csv_path = os.path.join(HAM_10000_path, "HAM10000_metadata.csv")
+    HAM_df = pd.read_csv(HAM_csv_path)
+
+    # Delete duplicate images in ISIC-19
+    ISIC_meta = pd.read_csv(os.path.join(ISIC_2019_path, "ISIC_2019_Training_Metadata.csv"))
+    barcelona_list = [i for i in ISIC_meta["lesion_id"].dropna() if "BCN" in i]
+    barcelona_core = ISIC_meta[ISIC_meta["lesion_id"].isin(barcelona_list)]
+
+    core_2019 = ISIC_df[ISIC_df["image"].isin(barcelona_core["image"])]
+    core_2019.to_csv(os.path.join(ISIC_2019_path, "ISIC_2019_core.csv"), mode="w")
+
+    # Split rosendahl and vienna in HAM10000
+    rosendahl_data = HAM_df[HAM_df["dataset"] == "rosendahl"]
+    rosendahl_data.to_csv(os.path.join(HAM_10000_path, "HAM_rosendahl.csv"), mode="w")
+    vienna_data = HAM_df[HAM_df["dataset"] != "rosendahl"]
+    vienna_data.to_csv(os.path.join(HAM_10000_path, "HAM_vienna.csv"), mode="w")
+
+    # Load PAD-UFES-20
+    PAD_UFES_20_path = f"{data_path}/PAD-UFES-20"
+    PAD_UFES_20_csv_path = os.path.join(PAD_UFES_20_path, "metadata.csv")
+    PAD_UFES_20 = pd.read_csv(PAD_UFES_20_csv_path)
+
+    # Load Derm7pt
+    Derm7pt_path = f"{data_path}/Derm7pt"
+    Derm7pt_csv_path = os.path.join(Derm7pt_path, "meta/meta.csv")
+    Derm7pt = pd.read_csv(Derm7pt_csv_path)
+
+    derm7pt_labelmap = {
```

Comment:
Correct me if I'm wrong, but it looks like this dictionary isn't used here and is then just redefined below? So we could just delete this definition.

Reply:
These are the label differences between the original dataset and the FL dataset, I have altered the code so this is more clear